src/kv/apply_changes.h

```diff
@@ -37,20 +37,26 @@ namespace kv
   // sets to their underlying Maps. Calls f() at most once, iff the writes are
   // applied, to retrieve a unique Version for the write set and return the max
   // version which can have a conflict with the transaction.
+  //
+  // The track_conflicts parameter tells the store if it needs to track the last
+  // read version for every key. This is required for backup execution as
+  // described at the top of tx.h
 
   static inline std::optional<std::tuple<Version, Version>> apply_changes(
     OrderedChanges& changes,
-    std::function<Version()> f,
+    std::function<std::tuple<Version, Version>(bool)> f,
```

Comment:
It's quite hard to tell what f is here, I think a better name and a named type alias would be nice.